src/components/ToDoList.js

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import ToDoListItem from "./ToDoListItem";
+import { Button } from "@cdk-uip/react-button";
+
+
+class ToDoList extends React.Component {
+
+    constructor(props){
+        super(props);
+        var items = [];
+        items.push({
+            index: 0, 
+            value: "",
+            checked: false,
+            disabled: false
+        });
+
+        this.state = {
+            items: items
+        }
+        this.onValueChange = this.onValueChange.bind(this);
+        this.onChecked = this.onChecked.bind(this);
+        this.addItem = this.addItem.bind(this);
+        this.deleteItem = this.deleteItem.bind(this);
+    }
+
+    
+    onValueChange(index, newValue){
+        let itm = this.state.items;
+        itm[index] = {index: index, value: newValue, checked: false, disabled: false};
```

Comment:
Instead of copying over every value, just update the value you need like: 

`itm[index].value = newValue`